elearning-frontend/src/components/ProfileSettings.js

```diff
@@ -39,32 +40,31 @@ const ProfileSettings = (props) => {
                     token: usertoken
                 }
             }).then(response => {
-                
-                if (response.data.user.profile.avatar !== null && response.data.user.profile.avatar != "") {
-                    setAvatar(response.data.user.profile.avatar);
-                    setProfilepic('http://127.0.0.1:8000/uploads/avatar/' + response.data.user.profile.avatar);
-                }
+                setUser({
+                    profilepic: (response.data.user.profile.avatar !== null && response.data.user.profile.avatar != "" ? 'http://127.0.0.1:8000/uploads/avatar/' + response.data.user.profile.avatar : 'https://st3.depositphotos.com/4111759/13425/v/600/depositphotos_134255710-stock-illustration-avatar-vector-male-profile-gray.jpg'),
+                    avatar: (response.data.user.profile.avatar !== null && response.data.user.profile.avatar != "" ? response.data.user.profile.avatar : null),
+                    firstname: response.data.user.profile.firstname,
+                    middlename: response.data.user.profile.middlename,
+                    lastname: response.data.user.profile.lastname,
+                    sex: response.data.user.profile.sex,
+                    birthdate: response.data.user.profile.birthdate,
+                    phone: response.data.user.profile.phone,
+                    address: response.data.user.profile.address,
+                    email: response.data.user.email,
+                    username: response.data.user.username,
+                    password: response.data.user.password,
+                })
```

Comment:
```suggestion
                setUser({response.data.user.profile})
                
                //or
                
                setUser({...user, ...response.data.user.profile})
```

- [ ] you can set the user object to user state like this. the one you applied is so costly and manual
- [ ] password is included here? You should not get the password from backend, its confidential

Reply:
User retrieval is updated @chevyB